src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -765,3 +823,21 @@ def alertDeletablePU(self, workflowName, containerName, ruleList):
         self.sendAlert(alertName, alertSeverity, alertSummary, alertDescription,
                        service=self.alertServiceName, endSecs=self.alertExpiration)
         self.logger.critical(alertDescription)
+
+    def alertStatusAdvanceExpired(self, wflow, additionalInfo=None):
+        """
+        Send alert notifying that a workflow has spend too much time in a given status
+        :param wflow: MSRuleCleanerWorkflow instance
+        :return: none
+        """
+        alertName = "{}: Stale Workflow: {}".format(self.alertServiceName, wflow['RequestName'])
+        alertSeverity = "high"
+        alertSummary = "[MSRuleCleaner] Found a stale workflow."
+        alertDescription = "\nWorkflow: {} has exceeded the Status Advance Timeout of: {} for status: {}.".format(wflow['RequestName'])
```

Comment:
Error, too few arguments.
And for the line below, I am not sure whether we have limit of number of characters for these alerts. The `additionalInfo` might be too much information for the alert.

Reply:
> Error, too few arguments.

I'll fix that.

> The additionalInfo might be too much information for the alert.

Well, I think it is actually important. It contains  the bit allowing you to distinguwish all the possible usecases which could trigger the alarm. this would shorten the debugging process significantly.

Reply:
Ok, but please check with Valentin and/or the CMS Monitoring on whether there are any limitations in the alert size (or any of their attributes, e.g. summary, description, etc). I remember hitting it back in the days of 100s of rule numbers in one of these alert attributes.